datastacks/utils.py

```diff
@@ -0,0 +1,27 @@
+from enum import Enum
+from jinja2 import Environment, FileSystemLoader, Undefined
+import yaml
+from pathlib import Path
+
+
+def render_template_components(config: dict, template_source_path: str, target_dir: str) -> None:
+    """Renders all templates within a given path with provided config, and saves results into a new target path,
+    while maintaining folder structure and removing jinja file extensions.
+
+    Args:
+        config: Dict of config containing required templating params
+        template_source_path: path containing templates to be rendered
+        target_dir: Directory to render templates into
+    """
+    Path(target_dir).mkdir(parents=True, exist_ok=True)
+    templateLoader = FileSystemLoader(searchpath=str(Path(template_source_path).absolute()))
+    templateEnv = Environment(loader=templateLoader, autoescape=True)
```

Comment:
Do we need autoescape? I tested the templates without this option, I'm not sure whether it is required and whether it changes something for our templates.

Reply:
Thanks, in that case it's better to keep it. I looked into that and it would have an effect if we'd use for example `&` in values passed in the config. But I don't see why we'd have to do it.